### PR TITLE
[debugging utils] minor doc improvements

### PR DESCRIPTION
**Pull request by stas00**
This PR includes various minor doc improvements.

@sgugger

### CODE REVIEW

File: src/transformers/debug_utils.py [hunk 1]
```diff
@@ -104,12 +107,19 @@ class DebugUnderflowOverflow:
     fast-forward right to that area.
 
 
+    Early stopping:
```
Comment (sgugger) [inline src/transformers/debug_utils.py:110]: It's missing something here, no?

**Review by sgugger — APPROVED** [review]
LGTM, thanks!

File: src/transformers/debug_utils.py (hunk 1 above)
Comment (stas00) [reply src/transformers/debug_utils.py:110]: What do you feel is missing?

I'm just sort of trying to delineate independent sections and I'm not sure how to do it inside a docstring. Any suggestions?

The "Early stopping" is an independent feature to either of the mode 1 or mode 2. and I didn't want it to "stick" to the mode 2 section.

Suggestions?

Comment (sgugger) [reply src/transformers/debug_utils.py:110]: I didn't understand the sentence below was in the same section since indent was the same, but this works.

Comment (stas00) [reply src/transformers/debug_utils.py:110]: Do you have any suggestions on how to create sub-sections inside the docstring?

Comment (sgugger) [reply src/transformers/debug_utils.py:110]: Not really, I only now how to do that in the rst.

**Pull request merged**